docs/design/services/dynamodb/high-level-library/prototype/option-1/sync/Prototype.java

```diff
@@ -0,0 +1,1025 @@
+
+    /**
+     * The item that contains the attribute being converted.
+     */
+    Item parent();
+
+    /**
+     * The schema of the {@link #parent()}.
+     */
+    ItemSchema parentSchema();
+}
+
+/**
+ * The result of invoking {@link DynamoDbDocumentClient#listTables()}.
+ */
+@ThreadSafe
+public interface ListTablesResponse {
+    /**
+     * A lazily-populated iterator over all tables in the current region. This may make multiple service calls in the
+     * background when iterating over the full result set.
+     */
+    SdkIterable<Table> tables();
+}
+
+/**
+ * A DynamoDB table, containing a collection of {@link Item}s.
+ *
+ * Currently supported operations:
+ * <ul>
+ *     <li>Writing objects with {@link #putItem(Item)} and {@link #putObject(Object)}</li>
+ *     <li>Reading objects with {@link #getItem(Item)}} and {@link #getObject(Object)}</li>
+ *     <li>Accessing the current table configuration with {@link #metadata()}.</li>
+ *     <li>Creating new indexes with {@link #createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest)}.</li>
+ * </ul>
+ *
+ * The full version will all table operations, including Query, Delete, Update, Scan, etc.
+ */
+@ThreadSafe
+public interface Table {
+    /**
+     * Retrieve the name of this table.
+     */
+    String name();
+
+    /**
+     * Invoke DynamoDB to retrieve the metadata for this table.
+     */
+    TableMetadata metadata();
+
+    /**
+     * Invoke DynamoDB to create a new global secondary index for this table.
+     *
+     * Usage Example:
+     * <code>
+     *     try (DynamoDbDocumentClient client = DynamoDb.documentClient()) {
+     *         ProvisionedCapacity indexCapacity = ProvisionedCapacity.builder()
+     *                                                                .readCapacity(5)
+     *                                                                .writeCapacity(5)
+     *                                                                .build();
+     *
+     *         KeySchema indexKeys = KeySchema.builder()
+     *                                        .putKey("extra-partition-key", ItemAttributeIndexType.PARTITION_KEY)
+     *                                        .build();
+     *
+     *         Table table = client.getTable("my-table");
+     *
+     *         table.createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest.builder()
+     *                                                                           .indexName("my-new-index")
+     *                                                                           .provisionedCapacity(tableCapacity)
+     *                                                                           .keySchema(tableKeys)
+     *                                                                           .build());
+     *     }
+     * </code>
+     */
+    CreateGlobalSecondaryIndexResponse createGlobalSecondaryIndex(CreateGlobalSecondaryIndexRequest createRequest);
```

Comment:
Can we have builder consumer overload :)

Reply:
We'd add consumer-builders everywhere in the real solution. I left them out to keep the design focused on the methods themselves.